astropy/convolution/tests/test_convolve_fft.py

```diff
@@ -987,7 +987,7 @@ def test_convolve_fft_boundary_wrap_error(error_kwarg):
         convolve_fft(x, y, boundary="wrap", **error_kwarg)
     assert (
         str(err.value)
-        == f"With boundary='wrap', {list(error_kwarg.keys())[0]} cannot be enabled."
+        == f"With boundary='wrap', {next(iter(error_kwarg.keys()))} cannot be enabled."
```

Comment:
`list(error_kwarg.keys())[0]` is easier to understand.

Reply:
Of course the nicest would be if `.keys()` were indexable, a feature IMO base Python should add.

Reply:
Yeah, actual code is ok but I would prioritize readability in a test unless it is testing big data. Otherwise, it is pre-mature optimization.

Reply:
In this case a more aggressive refactoring would produce a better result:
```python
@pytest.mark.parametrize("error_kwarg", ["psf_pad", "fft_pad", "dealias"])
def test_convolve_fft_boundary_wrap_error(error_kwarg):
    x = np.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]], dtype=">f8")
    y = np.array([[1.0]], dtype=">f8")
    assert (convolve_fft(x, y, boundary="wrap") == x).all()
    with pytest.raises(
        ValueError,
        match=rf"^With boundary='wrap', {error_kwarg} cannot be enabled\.$"
    ):
        convolve_fft(x, y, boundary="wrap", **{error_kwarg: True})
```

Reply:
As long as @keflavich is okay with the aggressive refactoring.

Reply:
yeah, totally.  I think this new version is... at least marginally clearer?  I'm happy to see new-to-me pytest features being used!

Reply:
Done!

Reply:
It is worth pointing out that when using the CLI it is currently not at all clear what `error_kwarg0` and others are supposed to be:
```
$ pytest --verbose --no-header astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 96 from PyObject
Internet access disabled
========================================= test session starts =========================================
collected 3 items                                                                                     

astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error[error_kwarg0] PASSED [ 33%]
astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error[error_kwarg1] PASSED [ 66%]
astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error[error_kwarg2] PASSED [100%]

========================================== 3 passed in 0.38s ==========================================
```
With the updated test the output is easier to understand:
```
$ pytest --verbose --no-header astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 96 from PyObject
Internet access disabled
========================================= test session starts =========================================
collected 3 items                                                                                     

astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error[psf_pad] PASSED [ 33%]
astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error[fft_pad] PASSED [ 66%]
astropy/convolution/tests/test_convolve_fft.py::test_convolve_fft_boundary_wrap_error[dealias] PASSED [100%]

========================================== 3 passed in 0.39s ==========================================

```